### ISSUE TITLE
support for easily jump to next/previous file

### ISSUE THREAD

**Issue by mfulton26** [microsoft/vscode-pull-request-github#4424]
I would like to be able to quickly iterate through changed files. e.g. One project I work on regularly has visual snapshots checked into the repository. When things change visually it is usually just a handful of snapshots but sometimes lots of images get updated. (e.g. We change global padding on our text inputs across many views.)

I would like to be able to press a button or a keyboard shortcut to go to the next file, etc.

Personally I would also like to be able to mark a file as viewed through a keyboard shortcut or even be able to mark the current file as viewed and automatically jump to the next file not marked as viewed.

**Comment by alexr00** [comment]
The command "Go to Next Diff in Pull Request" will go to the next diff across files. The command "Mark File as Viewed" will mark the active file as viewed. You can set keyboard shortcuts for both of these commands.

**Comment by alexr00** [comment]
What's missing is a way to run one command to mark as viewed and go to the next file.

**Comment by mfulton26** [comment]
hmmmm… I don't see commands named "Go to Next Diff in Pull Request" or "Mark File as Viewed" 🤔

**Comment by mfulton26** [comment]
I would like a "Go to Previous Diff in Pull Request" too (I don't see that one currently).

**Comment by alexr00** [comment]
"Go to Previous Diff in Pull Request" is already available in the pre-release build. We do indeed only who the commands when the file is a text diff. I can look at expanding this to other diffs in February.

**Comment by mfulton26** [comment]
Awesome. Thank you.

**Comment by diminutivesloop** [comment]
The existing commands are helpful but it would be great if there was a button in the toolbar for moving quickly to the next or previous diff/file in a PR without marking it as read. Sometimes I like to start a PR review by quickly looking through the files before reviewing them in depth and so I don't always want to mark them as read before moving on.

**Comment by TristenFielding** [comment]
Actually, the button that marks the file as view should not only close the current file but then open the next file.

**Comment by jjspace** [comment]
> Actually, the button that marks the file as view should not only close the current file but then open the next file.

Just cross linking because I really wish this whole workflow worked nicer together

- https://github.com/microsoft/vscode-pull-request-github/issues/5092
- https://github.com/microsoft/vscode-pull-request-github/issues/5537

**Comment by thorstenstark** [comment]
> > Actually, the button that marks the file as view should not only close the current file but then open the next file.
> 
> Just cross linking because I really wish this whole workflow worked nicer together
> 
> * [Option to *not* close file marked as viewed via keybinding #5092](https://github.com/microsoft/vscode-pull-request-github/issues/5092)
> * [Automatically advance to next file after marking file as viewed #5537](https://github.com/microsoft/vscode-pull-request-github/issues/5537)

I totally agree to @jjspace here. I would like to navigate a pull request only with keyboard shortcuts. Jumping to the next file when the current one is marked as viewed and ideally jumping to "Description" after the last file was marked as viewed.

**Comment by npeeples** [comment]
I'm not positive if this is related to this or not, but if I bind to `pr.goToNextDiffInPr`, I get an error "Current editor isn't a diff editor."

I use vscodevim, and was hoping to quickly navigate between the diff sections, but it seems like either I can't the 2 to play nicely, or there's something about this command?

<img width="792" height="63" alt="Image" src="https://github.com/user-attachments/assets/ba724ccf-13af-4c87-aab5-a7e54b6a075c" />

Previous is huge too to round out the workflow.

**Comment by alexr00** [comment]
Was your active editor a diff editor?

**Comment by npeeples** [comment]
I'm not 100% positive, but I believe so.

tab title "Changes in Pull Request #167 (5 files)"

I get there by clicking the Open Changes in the Pull Requests side window.

<img width="950" height="398" alt="Image" src="https://github.com/user-attachments/assets/bbf46a0a-69d1-4e2e-94bd-6959361f7f6b" />

**Comment by alexr00** [comment]
Thank you for the details! The go to next pr diff commands don't support the multi-diff editor. I'll open an issue to track adding that support.

**Comment by alexr00** [comment]
https://github.com/microsoft/vscode-pull-request-github/issues/7608
